consensus/misc/kip71.go

```diff
@@ -11,6 +11,9 @@ import (
 )
 
 func VerifyMagmaHeader(parentHeader, header *types.Header, kip71Config *params.KIP71Config) error {
+	if parentHeader == nil {
+		return fmt.Errorf("parent header is empty. (header number = %v)", header.Number)
```

Comment:
Is it impossible to avoid nil header fundamentally? Apart from the question, verification seems to be necessary.

Reply:
Yes, if we consider a malicious node, which is a base entity in P2P. I didn't explore yet the appearance of nil header for a case of benign behaviors.

Reply:
I think it's better to throw an error if the parent header verification fails. Can you tell me why it prints an error instead of throwing it? 
When parents[len(parents)-1] is nil in verifyCascadingFields function, it return a consensus.ErrUnknownAncestor. 
(https://github.com/klaytn/klaytn/blob/dev/consensus/istanbul/backend/engine.go#L266)

Reply:
I agree @henry-will's suggestion.

Insertchain() In the blockchain.go controls the `ErrUnknownAncestor` error like following
```go
case err == consensus.ErrUnknownAncestor && bc.futureBlocks.Contains(block.ParentHash()):
   bc.futureBlocks.Add(block.Hash(), block)
   stats.queued++
   continue
```

Reply:
`fmt.Errorf` returns `error` type to avoid panic, not just printing. The error `ErrUnknownAncestor` seems appropriate return type. Do you guys agree to return `ErrUnknownAncestor`?

Reply:
@hyunsooda Yes I agree. Returning the unknown error will make BAD BLOCK situation.

Reply:
Thanks, Changed. PTAL.